### DISCUSSION TITLE
Spinning e-puck for an angle: not working

### DISCUSSION THREAD

**Discussion by DisaacD** [cyberbotics/webots#6445] [General]
Hi,
I used the code below to spin e-puck for an angle theta; where
ROBOT_ANGULAR_SPEED_IN_DEGREES= 278.237392796. The hardcopy document I have explained how the mathematical formula was derived. However, when I run the simulation, the robot rotates for  a much lower angle than theta. Any advice,please?

'''

	{

		double duration = abs(theta) / ROBOT_ANGULAR_SPEED_IN_DEGREES;

     	// run the simulator
		double start_time = wb_robot_get_time();

		if (theta > 0)
		{
			motorRotateLeft();
		}
		else if (theta < 0)
		{
			motorRotateRight();
		}
		
		do
		{
			step();
		}
		while (wb_robot_get_time() < start_time + duration);
	}
'''

**Comment by omichel** [comment]
Yes, this is because of inertia. The motors are not strong enough to reach the maximum speed instantly. Therefore the actual average speed is lower than the maximum speed.
